exercises/simple-linked-list/simple-linked-list.spec.js

```diff
@@ -17,7 +17,7 @@ describe('simple-linked-list', function () {
       expect(el).toBeDefined();
 
       expect(function () {
-        var el = Element(1);
+        el = Element(1);
```

Comment:
So, there are a couple of problems here:

1. You've declared `el` as a global variable by not using the `var` keyword.  That's contrary to best practices and should only be done for a very good reason.
2. You don't really need `el` at all.  To get your exception to throw, all you need to do is invoke the constructor, you don't need to assign the result to anything.

Reply:
Hi @matthewmorgan, thank you for working with me through this. 

>You've declared el as a global variable by not using the var keyword. That's contrary to best practices and should only be done for a very good reason.
Oh really? I did not know that declaring something without using var makes it global in javascript. That being said, wouldn't this just be assigning a new value to the local _var el_ declared on line 16?

> You don't really need el at all. To get your exception to throw, all you need to do is invoke the constructor, you don't need to assign the result to anything.
Just tried this out which makes Lint throw an error saying "_A function with a name starting with an uppercase letter should only be used as a constructor  new-cap_", which goes back to your other comments about how a constructor should be called i.e. with a new keyword. And I guess I see your point on how they are both checking for the same thing, but then how would we check for the case that calling the constructor without the new keyword throws an error in the Jasmine test but it doesn't cause Lint to err?

Reply:
OK, I have to retract what I said here:
1. regarding `el`-- yes, in this case you are right-- it was declared earlier, and this would count as a reassignment.  If you hadn't declared it already, you would get a global variable.  I was not looking back in the code when I made that assertion.
2. You're right about this-- you can just call `new MyConstructor()` as far as JS is concerned, but it violates a different linting rule we have set up.  So, the way around this is either to annotate allowing an unused variable, or allowing the use of `new` for side-effects.  Your choice.